### PR TITLE
Drive-by refactoring

### PR DESCRIPTION
**Pull request by Michael-F-Bryan**
These are a couple things I've picked up while reading through the `kurbo` source code.

I've tried to put each change into its own commit so they can be easily reverted if not desired.

### CODE REVIEW

**Comment by Michael-F-Bryan** [comment]
@raphlinus would you be able to have a look at this and tell me what you think?

The main area I'm concerned about is docs. I added the `#[deny(missing_docs)]` link, but that meant I needed to fill in the missing docs. In particular, my interpretation of arc radii (both SVG and `arc.rs`) may not be 100% correct.

**Comment by raphlinus** [comment]
Generally this looks good to me. The one thing I might push back on is the Debug derives on iterator types. I feel those are internal details of the functions that produce the iterators, so very unlikely to be useful. I'm also (perhaps overly sensitive) to compile time, as I hope this to be a basic, foundational crate, but admit to not having measured the impact of Debug derives.

File: src/arc.rs [hunk 1]
```diff
@@ -6,10 +6,16 @@ use std::f64::consts::{FRAC_PI_2, PI};
 /// A single arc segment.
 #[derive(Clone, Copy, Debug)]
 pub struct Arc {
+    /// The arc's centre point.
```
Comment (raphlinus) [inline src/arc.rs]: I believe this is right, but it's no longer fresh in my mind, and the logic was adapted from a number of sources (including lyon).

File: src/bezpath.rs [hunk 2]
```diff
@@ -20,18 +20,26 @@ pub struct BezPath(Vec<PathEl>);
 /// A valid path has `Moveto` at the beginning of each subpath.
 #[derive(Clone, Copy, Debug)]
 pub enum PathEl {
+    /// Move directly to the point without drawing anything.
```
Comment (raphlinus) [inline src/bezpath.rs]: Maybe add ", starting a new subpath."

File: src/svg.rs [hunk 3]
```diff
@@ -10,11 +12,18 @@ use crate::{Arc, BezPath, ParamCurve, PathEl, PathSeg, Point, Vec2};
...
+    /// The arc's end point.
     pub to: Point,
+    /// The arc's radii, where the vector's x-component is the radius in the
+    /// positive x direction after applying `x_rotation`.
     pub radii: Vec2,
+    /// How much the arc is rotated, in radians.
     pub x_rotation: f64,
+    /// Does this arc sweep through more than π radians?
     pub large_arc: bool,
+    /// Determines if the arc should begin moving at positive angles.:wq
```
Comment (raphlinus) [inline src/svg.rs]: Vi user? :)

File: src/svg.rs [hunk 4]
```diff
@@ -47,141 +56,156 @@ impl BezPath {
...
+                )?,
+                PathEl::ClosePath => write!(writer, "Z")?,
             }
         }
-        String::from_utf8(result).unwrap()
+
+        Ok(())
     }
 
+    /// Try to parse a bezier path from an SVG path element.
```
Comment (raphlinus) [inline src/svg.rs]: I'm wondering if we should say that we don't guarantee that all SVG paths are handled correctly (this was discussed in #78), in the interest of full disclosure. It's intended for cases (such as vector icons) where the user controls the source of paths, and is not intended as a replacement for a general, robust SVG parser.

File: src/svg.rs (hunk 3 above)
Comment (Michael-F-Bryan) [reply src/svg.rs]: Guilty.

**Comment by Michael-F-Bryan** [comment]
@raphlinus I believe I've addressed all the points you raised.

I was tossing up whether to add a `FromStr` impl for `BezPath` which just defers to `BezPath::from_svg()`. Eventually I decided against it because that implies there's only one way to encode a `BezPath` as text (SVG), but from what I can tell a `BezPath` is just a list of draw instructions and has nothing to do with SVG... Does that sound correct?

**Comment by cmyr** [comment]
@Michael-F-Bryan your last point is correct, there are many different possible ways a bezier might be represented in text outside of the SVG case, so I think `FromStr` would be overstepping.

**Comment by Michael-F-Bryan** [comment]
@raphlinus is there anything else I need to do for this PR?

**Pull request merged**